app/2fa/server/methods/callWithTwoFactorRequired.ts

```diff
@@ -0,0 +1,17 @@
+import { Meteor } from 'meteor/meteor';
+
+import { checkCodeForUser } from '../code';
+import { methodOptions } from '../twoFactorRequired';
+
+Meteor.methods({
+	callWithTwoFactorRequired({ code, ddpMethod, method, params }: {code: string; ddpMethod: string; method: string; params: any[]}) {
+		if (!this.userId) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'callWithTwoFactorRequired' });
+		}
+
+		checkCodeForUser({ user: this.userId, code, method, options: methodOptions[ddpMethod], connection: this.connection || undefined });
+
+		this.twoFactorChecked = true;
+		return Meteor.server.method_handlers[ddpMethod].apply(this, params);
```

Comment:
how safe is this `Meteor.server.method_handlers[ddpMethod].apply`? should we have a `TODO` for things we should check on every Meteor update?

Reply:
Now I can replace this by a meteor call since we tweaked the method invocation